### PR TITLE
iOS: Add new method to connect to a SSIDPrefix protected only once, Android: Try to address issue #303

### PR DESCRIPTION
**Pull request by notjulian**
- iOS: Add new method to connect to a SSIDPrefix protected only once.
- Android: Add timeout for removeWifiNetwork and disconnect
- Android: Try to address issue #303.

For issue #303, I've tested it on a Pixel 7 running Android 14, as well as on other brands.
The fix is taken from [WiFiFlutter](https://github.com/flutternetwork/WiFiFlutter/pull/309) [concurrent-connections](https://developer.android.com/about/versions/12/behavior-changes-12#concurrent-connections)
Thanks to: @Martichou, @darhaywa and @daadu

Code example (for a hotspot without Internet), first connect, then forceWifiUsage true, before disconnect forceWiFiUsage to false

```
//To connect
await WifiManager.connectToProtectedWifiSSID({
                  ssid: WifiNetToUse,
                  password: pwdWifi,
                  isWEP: false,
                  isHidden: false,
                  timeout: 15,
                })
                  .then(async () => {
                    if (!isIOS)
                      await WifiManager.forceWifiUsageWithOptions(true, {
                        noInternet: true,
                      })
                  })
                  .catch(e => {
                    console.log(e.message, e.code)
                    error = e.code
                  })

//To disconnect
 await WifiManager.forceWifiUsageWithOptions(false, {
            noInternet: false,
})
await WifiManager.isRemoveWifiNetwork(currentSSID)
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "react-native-wifi-reborn",
-  "version": "4.12.1",
```
Comment (JuanSeBestia) [inline package.json:3]: You don't need update the package manually, semantic-release would do automatically

**Comment by JuanSeBestia** [comment]
I guess it has some conflicts for this MR
https://github.com/JuanSeBestia/react-native-wifi-reborn/pull/355

**Comment by notjulian** [comment]
> I guess it has some conflicts for this MR #355

sorry where?
thanks

**Comment by smartmedev** [comment]
Hi @notjulian why did you add timeout for removeWifiNetwork and disconnect on Android?
What is the purpose?
I've found that you userd DisconnectCallbackHolder.getInstance() on the disconnect() and removeWifiNetwork() timeout blocks and Android Studio tell us that this call requires API Level 29 when the library target is 21.
Screenshot 2024-05-31 alle 14 25 10

Can you please add a check like this if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) { } and use the old behavior elsewhere?

**Comment by notjulian** [comment]
> Hi @notjulian why did you add timeout for removeWifiNetwork and disconnect on Android? What is the purpose? I've found that you userd DisconnectCallbackHolder.getInstance() on the disconnect() and removeWifiNetwork() timeout blocks and Android Studio tell us that this call requires API Level 29 when the library target is 21. Screenshot 2024-05-31 alle 14 25 10
> 
> Can you please add a check like this if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) { } and use the old behavior elsewhere?

Hi,
is to avoid timeout on disconnect, I've updated the code

**Comment by smartmedev** [comment]
Hi @notjulian, sorry but what do you mean with "avoid timeout on disconnect"?

In @JuanSeBestia  library code on disconnect/removeWifiNetwork is there any timeout... you've added it on your pr, but I can't understand what is the purpose of adding a timeout on disconnect/removeWifiNetwork. Thank you

**Comment by notjulian** [comment]
> Hi @notjulian, sorry but what do you mean with "avoid timeout on disconnect"?
> 
> In @JuanSeBestia library code on disconnect/removeWifiNetwork is there any timeout... you've added it on your pr, but I can't understand what is the purpose of adding a timeout on disconnect/removeWifiNetwork. Thank you

If the disconnect take more time to complete, it will cancel and return

**Comment by gustavoabel** [comment]
@JuanSeBestia Hey, we already have 2 months of this PR, can we continue with the merge?

**Comment by JuanSeBestia** [comment]
FYI still "This branch cannot be rebased due to conflicts"

**Comment by notjulian** [comment]
@JuanSeBestia can you please check I've this:
` Able to merge. These branches can be automatically merged.`

Screenshot 2024-08-05 at 14 24 51

**Comment by JuanSeBestia** [comment]
@notjulian 
image

I can squash and merge if you want, but maybe it remove you as a commit owner, you can squash by you own if you want

**Comment by JuanSeBestia** [comment]
Mhe I will just merge it

**Comment by notjulian** [comment]
> Mhe I will just merge it, sorry for the delay.

thanks

**Comment by notjulian** [comment]
> @notjulian image
> 
> I can squash and merge if you want, but maybe it remove you as a commit owner, you can squash by you own if you want

yes can I do it or is too late now?

**Pull request merged**